lighthouse-core/test/report/v2/report-renderer-test.js

```diff
@@ -0,0 +1,177 @@
+
+function setupJsDomGlobals() {
+  global.document = jsdom.jsdom(TEMPLATES_FILE);
+  global.window = global.document.defaultView;
+  global.self = global.window;
+}
+
+function cleanupJsDomGlobals() {
+  global.document = undefined;
+  global.window = undefined;
+  global.self = undefined;
+}
+
+setupJsDomGlobals(); // Run jsdom setup code before report renderer is executed.
+require('../../../report/v2/report-renderer.js');
+
+describe('DOM', () => {
+  const window = self;
+  const DOM = window.DOM;
+
+  before(setupJsDomGlobals);
+  after(cleanupJsDomGlobals);
+
+  describe('setText', () => {
+    it('sets text of element', () => {
+      const el = document.createElement('div');
+      const result = DOM.setText(el, 'hello world');
+      assert.deepStrictEqual(result, el);
+      assert.equal(result.textContent, 'hello world');
+    });
+
+    it('does not fail when given null element argument', () => {
+      assert.doesNotThrow(() => {
+        const result = DOM.setText(null, 'hello world');
+        assert.strictEqual(result, null);
+      });
+    });
+  });
+
+  describe('addClass', () => {
+    it('adds classes to element', () => {
+      const el = document.createElement('div');
+      const result = DOM.addClass(el, 'class1', 'class2');
+      assert.ok(result.classList.contains, 'class1');
+      assert.ok(result.classList.contains, 'class2');
+    });
+
+    it('does not fail when given null element argument', () => {
+      assert.doesNotThrow(() => {
+        const result = DOM.addClass(null, 'class1');
+        assert.strictEqual(result, null);
+      });
+    });
+  });
+
+  describe('createElement', () => {
+    it('creates a simple element using default values', () => {
+      const dom = new DOM(document);
+      const el = dom.createElement('div');
+      assert.equal(el.localName, 'div');
+      assert.equal(el.className, '');
+      assert.equal(el.className, el.attributes.length);
+    });
+
+    it('creates an element from parameters', () => {
+      const dom = new DOM(document);
+      const el = dom.createElement('div', 'class1 class2', {title: 'title attr', tabindex: 0});
+      assert.equal(el.localName, 'div');
+      assert.equal(el.className, 'class1 class2');
+      assert.equal(el.getAttribute('title'), 'title attr');
+      assert.equal(el.getAttribute('tabindex'), '0');
+    });
+  });
+
+  describe('cloneTemplate', () => {
+    it('should clone a template', () => {
+      const dom = new DOM(document);
+      const clone = dom.cloneTemplate('#tmpl-lighthouse-audit-score');
+      assert.ok(clone.querySelector('.lighthouse-score'));
+    });
+
+    it('fails when template cannot be found', () => {
+      const dom = new DOM(document);
+      assert.throws(() => {
+        const clone = dom.cloneTemplate('#unknown-selector'); // eslint-disable-line no-unused-vars
+      });
+    });
+  });
+});
+
+describe('ReportRenderer V2', () => {
+  const window = self;
```

Comment:
isn't the global `window` already set to `self`? Or...something? The globals are confusing here (and only used to access `ReportRenderer`)